src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -39,17 +40,26 @@
 /* controls use of locks to enforce safe commutativity */
 bool AllModificationsCommutative = false;
 
+static uint64 shardForTransaction = INVALID_SHARD_ID;
+static bool callbackInstalled = false;
```

Comment:
Removing in favor of using the `shmem` callbacks to install the transaction callbacks.